### DISCUSSION TITLE
Have `poetry update` clearly denote the special status of 0.x versions?

### DISCUSSION THREAD

**Discussion by acdha** [python-poetry/poetry#8158] [Ideas]
I’m Currently, if you run `poetry update` and have any projects in your `pyproject.toml` which have the major version zero, Poetry's output will be misleading:

```bash
$ poetry update 
Updating dependencies
Resolving dependencies... (11.2s)

No dependencies to install or update
```

This is mentioned in the documentation for [dependency specification](https://python-poetry.org/docs/dependency-specification/) but it's not prominent and many people will end up on that page after they realize that they have a problem, not before.

It seems like this could be improved by having Poetry indicate when this happens and reminding the user - perhaps something along these lines:

```bash
$ poetry update 
Updating dependencies
Resolving dependencies... (11.2s)

No dependencies to install or update

Some pre-release dependencies were held back:

httpx 0.22.1 is installed. 0.23.0 is available but will not be updated until you run “poetry add httpx@latest".
unicorn 0.21.1 is installed. 0.22.3 is available but will not be updated until you run “poetry add uvicorn@latest".

See https://python-poetry.org/docs/dependency-specification/
```

Since `update` is already an extremely noisy operation with things like download progress bars the extra clarity seems like a good tradeoff.

**Comment by y2kbugger** [comment]
I agree, especially coming across this for the first time. 0.X.X versioned projects are often rapidly release new versions. I was getting behind with FastAPI, and ruff for example.

**Reply by shrektan** [reply]
Yes, it isn't delightful. I have to manually modify all the libraries with versions starting with "^0.x.x" to ">=0.x.x" to have the latest version of the packages, which is not very convenient and easy for users to ignore.
